// ra-richtext-tiptap/esm/RichTextInputToolbar.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import * as React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { RichTextInputLevelSelect } from './RichTextInputLevelSelect';
import { AlignmentButtons, FormatButtons, ListButtons, LinkButtons, QuoteButtons, ClearButtons, } from './RichTextInputDefaultButtons';
/**
 * A toolbar for the <RichTextInput>.
 * @param props The toolbar props.
 * @param {ReactNode} props.children The toolbar children, usually many <ToggleButton>.
 * @param {'small' | 'medium' | 'large'} props.size The default size to apply to the **default** children.
 *
 * @example <caption>Customizing the size</caption>
 * import { RichTextInput, RichTextInputToolbar } from 'ra-richtext-tiptap';
 * const MyRichTextInput = (props) => (
 *     <RichTextInput
 *         toolbar={<RichTextInputToolbar size="large" />}
 *         label="Body"
 *         source="body"
 *         {...props}
 *     />
 * );
 *
 * @example <caption>Customizing the children</caption>
 * import { RichTextInput, RichTextInputToolbar } from 'ra-richtext-tiptap';
 * const MyRichTextInput = ({ size, ...props }) => (
 *     <RichTextInput
 *         toolbar={(
 *             <RichTextInputToolbar>
 *                 <RichTextInputLevelSelect size={size} />
 *                 <FormatButtons size={size} />
 *                 <ListButtons size={size} />
 *                 <LinkButtons size={size} />
 *                 <QuoteButtons size={size} />
 *                 <ClearButtons size={size} />
 *             </RichTextInputToolbar>
 *         )}
 *         label="Body"
 *         source="body"
 *         {...props}
 *     />
 * );
 */
export var RichTextInputToolbar = function (props) {
    var classes = useStyles(props);
    var _a = props.size, size = _a === void 0 ? 'medium' : _a, _b = props.children, children = _b === void 0 ? (React.createElement(React.Fragment, null,
        React.createElement(RichTextInputLevelSelect, { size: size }),
        React.createElement(FormatButtons, { size: size }),
        React.createElement(AlignmentButtons, { size: size }),
        React.createElement(ListButtons, { size: size }),
        React.createElement(LinkButtons, { size: size }),
        React.createElement(QuoteButtons, { size: size }),
        React.createElement(ClearButtons, { size: size }))) : _b, rest = __rest(props, ["size", "children"]);
    return (React.createElement("div", __assign({ className: classes.root }, rest), children));
};
var useStyles = makeStyles(function (theme) { return ({
    root: {
        display: 'inline-flex',
        marginBottom: theme.spacing(1),
        alignItems: 'center',
        '& > *': {
            marginRight: theme.spacing(1),
        },
        '& > *:last-child': {
            marginRight: 0,
        },
    },
}); }, {
    name: 'RaRichTextInputToolbar',
});
